ms2query/spectrum_processing.py

```diff
@@ -0,0 +1,162 @@
+    """
+    settings = set_minimal_processing_defaults(**settings)
+    spectrum = normalize_intensities(spectrum)
+    spectrum = select_by_intensity(spectrum, intensity_from=settings["intensity_from"])
+    spectrum = select_by_mz(spectrum, mz_from=settings["mz_from"], mz_to=np.inf)
+    spectrum = require_peaks_below_mz(spectrum, n_required=settings["n_required_below_1000"],
+                                      max_mz=settings["max_mz_required"])
+    return spectrum
+
+def set_spec2vec_defaults(**settings: Dict[str, Union[int, bool]]) \
+        -> Dict[str, Union[int, bool]]:
+    """Set spec2vec default argument values"(where no user input is given)".
+
+    Args
+    ----------
+    **settings:
+        Change default settings
+    """
+    defaults = {"mz_from": 0,
+                "mz_to": 1000,
+                "n_required": 1,
+                "ratio_desired": 0.5,
+                "intensity_from": 0.001,
+                "n_max": 1000,
+                "loss_mz_from": 5.0,
+                "loss_mz_to": 200.0,
+                "intensity_weighting_power": 0.5,
+                "allowed_missing_percentage": 10.0,
+                }
+
+    # Set default parameters or replace by **settings input
+    for key in defaults:
+        if key not in settings:
+            settings[key] = defaults[key]
+    return settings
+
+
+def spectrum_processing_s2v(spectrum: SpectrumType,
+                            **settings: Dict[str, Union[int, bool]]) \
+    -> Union[SpectrumType, None]:
+    """Spectrum processing required for computing Spec2Vec scores.
+
+    Args:
+    ----------
+    spectrum:
+        Spectrum to process
+    mz_from
+        Set lower threshold for m/z peak positions. Default is 0.0.
+    mz_to
+        Set upper threshold for m/z peak positions. Default is 1000.0.
+    n_required
+        Number of minimal required peaks for a spectrum to be considered.
+    ratio_desired
+        Number of minimum required peaks. Spectra with fewer peaks will be set
+        to 'None'. Default is 1.
+    intensity_from
+        Set lower threshold for peak intensity. Default is 10.0.
+    n_max
+        Maximum number of peaks to be kept per spectrum. Default is 1000.
+    loss_mz_from
+        Minimum allowed m/z value for losses. Default is 0.0.
+    loss_mz_to
+        Maximum allowed m/z value for losses. Default is 1000.0.
+    intensity_weighting_power
+        Weighing of individual word (=peak) vectors.
+    allowed_missing_percentage
+        Allowed percentage of the entire spectrum that is allowed to not be
+        covered my the pretrained Spec2Vec model.
+    """
+    settings = set_spec2vec_defaults(**settings)
+    spectrum = reduce_to_number_of_peaks(spectrum,
+                                         n_required=settings["n_required"],
+                                         ratio_desired=settings["ratio_desired"],
+                                         n_max=settings["n_max"])
+    if spectrum is None:
```

Comment:
True, good point. It's not needed for this function, so I took it out (I didn't add a assert though, because it would anyway then fail with None).

Reply:
I think it would not fail anyway, since the reduce_to_number_of_peaks and add_losses are able to handle None as an input spectrum and it will return None again. See https://github.com/matchms/matchms/blob/master/matchms/filtering/reduce_to_number_of_peaks.py
So maybe good to still add the assert function.